From 4fe1b1458d21a4f254cd2866a316df540450de12 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 8 Oct 2017 04:04:27 +0300
Subject: [PATCH 1/1] Solus patch for various improvements or tweaks to GNOME
 Control Center.

- Add Solus branding. Launch solus-sc instead of gnome-software.
- Don't hide desktop items.
- Fix binary location of passwd.
- panels: Disable GDM specific functionality for now
- Add black background as an option. It's the simple things.
- panels: Change About to indicate OS version instead of GTK version, add GTK version below Device name.
---
 panels/background/bg-colors-source.c               |  1 +
 .../gnome-background-panel.desktop.in.in           |  1 -
 .../bluetooth/gnome-bluetooth-panel.desktop.in.in  |  1 -
 panels/color/gnome-color-panel.desktop.in.in       |  1 -
 panels/datetime/gnome-datetime-panel.desktop.in.in |  1 -
 panels/display/gnome-display-panel.desktop.in.in   |  1 -
 panels/info/cc-info-overview-panel.c               | 22 +++++++-------
 panels/info/gnome-default-apps-panel.desktop.in.in |  1 -
 .../info/gnome-info-overview-panel.desktop.in.in   |  1 -
 panels/info/gnome-info-panel.desktop.in.in         |  3 +-
 .../info/gnome-removable-media-panel.desktop.in.in |  1 -
 panels/info/info-overview.ui                       | 34 ++++++++++++----------
 panels/info/info.gresource.xml                     |  1 -
 panels/keyboard/gnome-keyboard-panel.desktop.in.in |  1 -
 panels/mouse/gnome-mouse-panel.desktop.in.in       |  1 -
 panels/network/gnome-network-panel.desktop.in.in   |  1 -
 panels/network/gnome-wifi-panel.desktop.in.in      |  1 -
 .../gnome-notifications-panel.desktop.in.in        |  1 -
 .../gnome-online-accounts-panel.desktop.in.in      |  1 -
 panels/power/gnome-power-panel.desktop.in.in       |  1 -
 panels/printers/gnome-printers-panel.desktop.in.in |  1 -
 panels/privacy/gnome-privacy-panel.desktop.in.in   |  1 -
 panels/region/gnome-region-panel.desktop.in.in     |  1 -
 panels/search/gnome-search-panel.desktop.in.in     |  1 -
 panels/sharing/gnome-sharing-panel.desktop.in.in   |  1 -
 panels/sound/data/gnome-sound-panel.desktop.in.in  |  1 -
 .../gnome-universal-access-panel.desktop.in.in     |  1 -
 .../data/gnome-user-accounts-panel.desktop.in.in   |  1 -
 panels/user-accounts/run-passwd.c                  |  2 +-
 panels/user-accounts/um-user-panel.c               | 10 +------
 panels/wacom/gnome-wacom-panel.desktop.in.in       |  1 -
 31 files changed, 33 insertions(+), 64 deletions(-)

diff --git a/panels/background/bg-colors-source.c b/panels/background/bg-colors-source.c
index afe58683e..6bd0401c6 100644
--- a/panels/background/bg-colors-source.c
+++ b/panels/background/bg-colors-source.c
@@ -52,6 +52,7 @@ struct {
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#475b52" },
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#425265" },
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#7a634b" },
+  { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#000000" },
 };
 
 static char *
diff --git a/panels/background/gnome-background-panel.desktop.in.in b/panels/background/gnome-background-panel.desktop.in.in
index 2b665f2ae..9ec0fdd38 100644
--- a/panels/background/gnome-background-panel.desktop.in.in
+++ b/panels/background/gnome-background-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center background
 Icon=preferences-desktop-wallpaper
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;X-GNOME-PersonalizationSettings;
 OnlyShowIn=GNOME;
diff --git a/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in b/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
index a0879dc45..22f0f1aca 100644
--- a/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
+++ b/panels/bluetooth/gnome-bluetooth-panel.desktop.in.in
@@ -6,7 +6,6 @@ Icon=bluetooth
 Exec=gnome-control-center bluetooth
 Terminal=false
 Type=Application
-NoDisplay=true
 Categories=GTK;GNOME;Settings;X-GNOME-NetworkSettings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-ConnectivitySettings;
 OnlyShowIn=GNOME;Unity;
 StartupNotify=true
diff --git a/panels/color/gnome-color-panel.desktop.in.in b/panels/color/gnome-color-panel.desktop.in.in
index 35b161a68..cde061fb3 100644
--- a/panels/color/gnome-color-panel.desktop.in.in
+++ b/panels/color/gnome-color-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center color
 Icon=preferences-color
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-Settings-Panel;HardwareSettings;X-GNOME-DevicesSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/datetime/gnome-datetime-panel.desktop.in.in b/panels/datetime/gnome-datetime-panel.desktop.in.in
index 446b23eda..16c475f1d 100644
--- a/panels/datetime/gnome-datetime-panel.desktop.in.in
+++ b/panels/datetime/gnome-datetime-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center datetime
 Icon=preferences-system-time
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;X-GNOME-DetailsSettings;
 OnlyShowIn=GNOME;
diff --git a/panels/display/gnome-display-panel.desktop.in.in b/panels/display/gnome-display-panel.desktop.in.in
index 3c8f32c17..0b86f62ea 100644
--- a/panels/display/gnome-display-panel.desktop.in.in
+++ b/panels/display/gnome-display-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center display
 Icon=preferences-desktop-display
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/info/cc-info-overview-panel.c b/panels/info/cc-info-overview-panel.c
index 3af58be15..707838bc4 100644
--- a/panels/info/cc-info-overview-panel.c
+++ b/panels/info/cc-info-overview-panel.c
@@ -58,9 +58,9 @@ typedef struct
   GtkWidget      *system_image;
   GtkWidget      *version_label;
   GtkWidget      *name_entry;
+  GtkWidget      *gv_label;
   GtkWidget      *memory_label;
   GtkWidget      *processor_label;
-  GtkWidget      *os_name_label;
   GtkWidget      *os_type_label;
   GtkWidget      *disk_label;
   GtkWidget      *graphics_label;
@@ -804,8 +804,8 @@ info_overview_panel_setup_overview (CcInfoOverviewPanel *self)
                             &priv->gnome_date);
   if (res)
     {
-      text = g_strdup_printf (_("Version %s"), priv->gnome_version);
-      gtk_label_set_text (GTK_LABEL (priv->version_label), text);
+      text = g_strdup_printf (_("%s"), priv->gnome_version);
+      gtk_label_set_text (GTK_LABEL (priv->gv_label), text);
       g_free (text);
     }
 
@@ -825,7 +825,7 @@ info_overview_panel_setup_overview (CcInfoOverviewPanel *self)
   g_free (text);
 
   text = get_os_name ();
-  gtk_label_set_text (GTK_LABEL (priv->os_name_label), text ? text : "");
+  gtk_label_set_text (GTK_LABEL (priv->version_label), text ? text : "");
   g_free (text);
 
   get_primary_disc_info (self);
@@ -834,9 +834,9 @@ info_overview_panel_setup_overview (CcInfoOverviewPanel *self)
 }
 
 static gboolean
-does_gnome_software_exist (void)
+does_solus_sc_exist (void)
 {
-  return g_file_test (BINDIR "/gnome-software", G_FILE_TEST_EXISTS);
+  return g_file_test (BINDIR "/solus-sc", G_FILE_TEST_EXISTS);
 }
 
 static gboolean
@@ -854,10 +854,10 @@ on_updates_button_clicked (GtkWidget           *widget,
   gchar **argv;
 
   argv = g_new0 (gchar *, 3);
-  if (does_gnome_software_exist ())
+  if (does_solus_sc_exist ())
     {
-      argv[0] = g_build_filename (BINDIR, "gnome-software", NULL);
-      argv[1] = g_strdup_printf ("--mode=updates");
+      argv[0] = g_build_filename (BINDIR, "solus-sc", NULL);
+      argv[1] = g_strdup_printf ("--update-view");
     }
   else
     {
@@ -917,9 +917,9 @@ cc_info_overview_panel_class_init (CcInfoOverviewPanelClass *klass)
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, system_image);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, version_label);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, name_entry);
+  gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, gv_label);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, memory_label);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, processor_label);
-  gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, os_name_label);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, os_type_label);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, disk_label);
   gtk_widget_class_bind_template_child_private (widget_class, CcInfoOverviewPanel, graphics_label);
@@ -939,7 +939,7 @@ cc_info_overview_panel_init (CcInfoOverviewPanel *self)
 
   priv->graphics_data = get_graphics_data ();
 
-  if (does_gnome_software_exist () || does_gpk_update_viewer_exist ())
+  if (does_solus_sc_exist () || does_gpk_update_viewer_exist ())
     {
       g_signal_connect (priv->updates_button, "clicked", G_CALLBACK (on_updates_button_clicked), self);
     }
diff --git a/panels/info/gnome-default-apps-panel.desktop.in.in b/panels/info/gnome-default-apps-panel.desktop.in.in
index 0154f34c7..9dd173495 100644
--- a/panels/info/gnome-default-apps-panel.desktop.in.in
+++ b/panels/info/gnome-default-apps-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center default-apps
 Icon=starred
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-Hidden;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;X-GNOME-DetailsSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/info/gnome-info-overview-panel.desktop.in.in b/panels/info/gnome-info-overview-panel.desktop.in.in
index 87427097b..8c35d4bc9 100644
--- a/panels/info/gnome-info-overview-panel.desktop.in.in
+++ b/panels/info/gnome-info-overview-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center info-overview
 Icon=help-about
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-Hidden;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;X-GNOME-DetailsSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/info/gnome-info-panel.desktop.in.in b/panels/info/gnome-info-panel.desktop.in.in
index 22714aafa..888e55c6a 100644
--- a/panels/info/gnome-info-panel.desktop.in.in
+++ b/panels/info/gnome-info-panel.desktop.in.in
@@ -2,11 +2,10 @@
 # Translators: Add soft hyphens to your translations so that the icon view won't clip your translations. See https://bugzilla.gnome.org/show_bug.cgi?id=647087#c13 for details
 _Name=De­tails
 _Comment=View information about your system
-Exec=gnome-control-center info
+Exec=gnome-control-center info-overview
 Icon=applications-system
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-AltHidden;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;X-GNOME-DetailsSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/info/gnome-removable-media-panel.desktop.in.in b/panels/info/gnome-removable-media-panel.desktop.in.in
index bc4675edc..c025a5c52 100644
--- a/panels/info/gnome-removable-media-panel.desktop.in.in
+++ b/panels/info/gnome-removable-media-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center removable-media
 Icon=media-removable
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-Hidden;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/info/info-overview.ui b/panels/info/info-overview.ui
index 219a83c4c..56b84f2ce 100644
--- a/panels/info/info-overview.ui
+++ b/panels/info/info-overview.ui
@@ -18,7 +18,9 @@
           <object class="GtkImage" id="system_image">
             <property name="visible">True</property>
             <property name="can_focus">False</property>
-            <property name="resource">/org/gnome/control-center/info/GnomeLogoVerticalMedium.svg</property>
+            <property name="pixel_size">192</property>
+            <property name="icon_name">start-here-solus</property>
+            <property name="icon_size">6</property>
           </object>
           <packing>
             <property name="expand">False</property>
@@ -67,12 +69,12 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="label5">
+              <object class="GtkLabel" id="label42">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">1</property>
-                <property name="label" translatable="yes">Memory</property>
-                <property name="mnemonic_widget">memory_label</property>
+                <property name="label" translatable="no">GNOME</property>
+                <property name="mnemonic_widget">gv_label</property>
                 <style>
                  <class name="dim-label"/>
                 </style>
@@ -83,12 +85,12 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="label6">
+              <object class="GtkLabel" id="label5">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">1</property>
-                <property name="label" translatable="yes">Processor</property>
-                <property name="mnemonic_widget">processor_label</property>
+                <property name="label" translatable="yes">Memory</property>
+                <property name="mnemonic_widget">memory_label</property>
                 <style>
                  <class name="dim-label"/>
                 </style>
@@ -99,11 +101,12 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="label14">
+              <object class="GtkLabel" id="label6">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">1</property>
-                <property name="label" translatable="yes">Graphics</property>
+                <property name="label" translatable="yes">Processor</property>
+                <property name="mnemonic_widget">processor_label</property>
                 <style>
                  <class name="dim-label"/>
                 </style>
@@ -114,12 +117,11 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="label7">
+              <object class="GtkLabel" id="label14">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">1</property>
-                <property name="label" translatable="yes" comments="To translators: this field contains the distro name and version">OS name</property>
-                <property name="mnemonic_widget">os_name_label</property>
+                <property name="label" translatable="yes">Graphics</property>
                 <style>
                  <class name="dim-label"/>
                 </style>
@@ -189,7 +191,7 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="memory_label">
+              <object class="GtkLabel" id="gv_label">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">0</property>
@@ -202,7 +204,7 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="processor_label">
+              <object class="GtkLabel" id="memory_label">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">0</property>
@@ -215,7 +217,7 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="graphics_label">
+              <object class="GtkLabel" id="processor_label">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">0</property>
@@ -228,7 +230,7 @@
               </packing>
             </child>
             <child>
-              <object class="GtkLabel" id="os_name_label">
+              <object class="GtkLabel" id="graphics_label">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="xalign">0</property>
diff --git a/panels/info/info.gresource.xml b/panels/info/info.gresource.xml
index ce49a67bd..4e99df293 100644
--- a/panels/info/info.gresource.xml
+++ b/panels/info/info.gresource.xml
@@ -5,6 +5,5 @@
     <file preprocess="xml-stripblanks">info-overview.ui</file>
     <file preprocess="xml-stripblanks">info-default-apps.ui</file>
     <file preprocess="xml-stripblanks">info-removable-media.ui</file>
-    <file>GnomeLogoVerticalMedium.svg</file>
   </gresource>
 </gresources>
diff --git a/panels/keyboard/gnome-keyboard-panel.desktop.in.in b/panels/keyboard/gnome-keyboard-panel.desktop.in.in
index 1d45e642e..63e999917 100644
--- a/panels/keyboard/gnome-keyboard-panel.desktop.in.in
+++ b/panels/keyboard/gnome-keyboard-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center keyboard
 Icon=input-keyboard
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/mouse/gnome-mouse-panel.desktop.in.in b/panels/mouse/gnome-mouse-panel.desktop.in.in
index 2e0abee49..f3bfb6941 100644
--- a/panels/mouse/gnome-mouse-panel.desktop.in.in
+++ b/panels/mouse/gnome-mouse-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center mouse
 Icon=input-mouse
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/network/gnome-network-panel.desktop.in.in b/panels/network/gnome-network-panel.desktop.in.in
index d0817ab29..1afb858b2 100644
--- a/panels/network/gnome-network-panel.desktop.in.in
+++ b/panels/network/gnome-network-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center network
 Icon=network-workgroup
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/network/gnome-wifi-panel.desktop.in.in b/panels/network/gnome-wifi-panel.desktop.in.in
index b37e650b8..f2ac1885d 100644
--- a/panels/network/gnome-wifi-panel.desktop.in.in
+++ b/panels/network/gnome-wifi-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center wifi
 Icon=network-wireless
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-ConnectivitySettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/notifications/gnome-notifications-panel.desktop.in.in b/panels/notifications/gnome-notifications-panel.desktop.in.in
index 1d9bc56a5..5f380f559 100644
--- a/panels/notifications/gnome-notifications-panel.desktop.in.in
+++ b/panels/notifications/gnome-notifications-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center notifications
 Icon=preferences-system-notifications
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;X-GNOME-PersonalizationSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in b/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in
index 271d9399a..6a06afa65 100644
--- a/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in
+++ b/panels/online-accounts/gnome-online-accounts-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center online-accounts
 Icon=goa-panel
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;X-GNOME-AccountSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/power/gnome-power-panel.desktop.in.in b/panels/power/gnome-power-panel.desktop.in.in
index 4eb577782..4a99f4e17 100644
--- a/panels/power/gnome-power-panel.desktop.in.in
+++ b/panels/power/gnome-power-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center power
 Icon=gnome-power-manager
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;HardwareSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/printers/gnome-printers-panel.desktop.in.in b/panels/printers/gnome-printers-panel.desktop.in.in
index 079264dd8..fc1a34cb2 100644
--- a/panels/printers/gnome-printers-panel.desktop.in.in
+++ b/panels/printers/gnome-printers-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center printers
 Icon=printer
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 # The X-GNOME-Settings-Panel is necessary to show in the main shell UI
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
diff --git a/panels/privacy/gnome-privacy-panel.desktop.in.in b/panels/privacy/gnome-privacy-panel.desktop.in.in
index ca8b53a67..0a7078ef7 100644
--- a/panels/privacy/gnome-privacy-panel.desktop.in.in
+++ b/panels/privacy/gnome-privacy-panel.desktop.in.in
@@ -7,7 +7,6 @@ Exec=gnome-control-center privacy
 Icon=preferences-system-privacy
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;X-GNOME-AccountSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/region/gnome-region-panel.desktop.in.in b/panels/region/gnome-region-panel.desktop.in.in
index 79d1e1246..81981cee4 100644
--- a/panels/region/gnome-region-panel.desktop.in.in
+++ b/panels/region/gnome-region-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center region
 Icon=preferences-desktop-locale
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;X-GNOME-PersonalizationSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/search/gnome-search-panel.desktop.in.in b/panels/search/gnome-search-panel.desktop.in.in
index 5679dbb9c..4614b9ff3 100644
--- a/panels/search/gnome-search-panel.desktop.in.in
+++ b/panels/search/gnome-search-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center search
 Icon=preferences-system-search
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalSettings;X-GNOME-PersonalizationSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/sharing/gnome-sharing-panel.desktop.in.in b/panels/sharing/gnome-sharing-panel.desktop.in.in
index c27d26610..284a09b6a 100644
--- a/panels/sharing/gnome-sharing-panel.desktop.in.in
+++ b/panels/sharing/gnome-sharing-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center sharing
 Icon=preferences-system-sharing
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;DesktopSettings;X-GNOME-Settings-Panel;X-GNOME-SystemSettings;X-GNOME-AccountSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/sound/data/gnome-sound-panel.desktop.in.in b/panels/sound/data/gnome-sound-panel.desktop.in.in
index 25b4f7ad6..78dda26c9 100644
--- a/panels/sound/data/gnome-sound-panel.desktop.in.in
+++ b/panels/sound/data/gnome-sound-panel.desktop.in.in
@@ -5,7 +5,6 @@ Exec=gnome-control-center sound
 Icon=multimedia-volume-control
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/universal-access/gnome-universal-access-panel.desktop.in.in b/panels/universal-access/gnome-universal-access-panel.desktop.in.in
index 5df235826..3362802e4 100644
--- a/panels/universal-access/gnome-universal-access-panel.desktop.in.in
+++ b/panels/universal-access/gnome-universal-access-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center universal-access
 Icon=preferences-desktop-accessibility
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;X-GNOME-SystemSettings;X-GNOME-Settings-Panel;X-GNOME-PersonalizationSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in b/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
index 5b35f2f78..bd17549c8 100644
--- a/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
+++ b/panels/user-accounts/data/gnome-user-accounts-panel.desktop.in.in
@@ -5,7 +5,6 @@ Exec=gnome-control-center user-accounts
 Icon=system-users
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=System;Settings;X-GNOME-Settings-Panel;X-GNOME-SystemSettings;X-GNOME-DetailsSettings;
 OnlyShowIn=GNOME;Unity;
diff --git a/panels/user-accounts/run-passwd.c b/panels/user-accounts/run-passwd.c
index 00239ce0f..52154f14c 100644
--- a/panels/user-accounts/run-passwd.c
+++ b/panels/user-accounts/run-passwd.c
@@ -150,7 +150,7 @@ spawn_passwd (PasswdHandler *passwd_handler, GError **error)
         gchar  **envp;
         gint    my_stdin, my_stdout, my_stderr;
 
-        argv[0] = "/usr/bin/passwd";    /* Is it safe to rely on a hard-coded path? */
+        argv[0] = "/bin/passwd";    /* Is it safe to rely on a hard-coded path? Josh: No, no it isn't. */
         argv[1] = NULL;
 
         envp = g_get_environ ();
diff --git a/panels/user-accounts/um-user-panel.c b/panels/user-accounts/um-user-panel.c
index c1875cb69..ac6c40de7 100644
--- a/panels/user-accounts/um-user-panel.c
+++ b/panels/user-accounts/um-user-panel.c
@@ -68,7 +68,6 @@ struct _CcUserPanelPrivate {
         GCancellable  *cancellable;
         GtkBuilder *builder;
         GtkWidget *notification;
-        GSettings *login_screen_settings;
 
         GtkWidget *headerbar_buttons;
         GtkWidget *main_box;
@@ -868,11 +867,7 @@ show_user (ActUser *user, CcUserPanelPrivate *d)
         /* Fingerprint: show when self, local, enabled, and possible */
         widget = get_widget (d, "account-fingerprint-button");
         label = get_widget (d, "account-fingerprint-label");
-        show = (act_user_get_uid (user) == getuid() &&
-                act_user_is_local_account (user) &&
-                (d->login_screen_settings &&
-                 g_settings_get_boolean (d->login_screen_settings, "enable-fingerprint-authentication")) &&
-                set_fingerprint_label (widget));
+        show = FALSE;
         gtk_widget_set_visible (label, show);
         gtk_widget_set_visible (widget, show);
 
@@ -1478,7 +1473,6 @@ cc_user_panel_init (CcUserPanel *self)
         g_object_unref (provider);
 
         d->headerbar_buttons = get_widget (d, "headerbar-buttons");
-        d->login_screen_settings = settings_or_null ("org.gnome.login-screen");
 
         d->password_dialog = um_password_dialog_new ();
         button = get_widget (d, "user-icon-button");
@@ -1497,8 +1491,6 @@ cc_user_panel_dispose (GObject *object)
         g_cancellable_cancel (priv->cancellable);
         g_clear_object (&priv->cancellable);
 
-        g_clear_object (&priv->login_screen_settings);
-
         if (priv->um) {
                 g_signal_handlers_disconnect_by_data (priv->um, priv);
                 priv->um = NULL;
diff --git a/panels/wacom/gnome-wacom-panel.desktop.in.in b/panels/wacom/gnome-wacom-panel.desktop.in.in
index ce98b2e27..01dbe750d 100644
--- a/panels/wacom/gnome-wacom-panel.desktop.in.in
+++ b/panels/wacom/gnome-wacom-panel.desktop.in.in
@@ -6,7 +6,6 @@ Exec=gnome-control-center wacom
 Icon=input-tablet
 Terminal=false
 Type=Application
-NoDisplay=true
 StartupNotify=true
 Categories=GNOME;GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;X-GNOME-DevicesSettings;
 OnlyShowIn=GNOME;Unity;
-- 
2.16.1

